lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -177,6 +189,192 @@ impl EnsoLexer {
 }
 
 
+// === Comments ===
+
+/// The set of rules for lexing Enso comments.
+#[allow(dead_code)]
+impl EnsoLexer {
+
+    // === Disable Comments ===
+
+    /// Triggered when a disable comment is encountered.
+    fn on_begin_disable_comment<R:LazyReader>(&mut self, _reader:&mut R){
+        trace!(self.logger,"Comment::on_begin_disable_comment");
+        let offset          = self.offset.consume();
+        let disable_comment = self.disable_comment;
+        self.comment_state.set_offset(offset);
+        self.push_state(disable_comment);
+    }
+
+    /// Accumulate the disable comment contents.
+    fn on_build_disable_comment<R:LazyReader>(&mut self, _reader:&mut R){
+        trace!(self.logger,"Comment::on_build_disable_comment");
+        let current = self.consume_current();
+        self.comment_state.append_to_line(current);
+    }
+
+    /// Triggered when a disable comment is ended.
+    fn on_end_disable_comment<R:LazyReader>(&mut self, line_end:token::LineEnding, reader:&mut R) {
```

Comment:
passing here the reader seems like a very strange design to me. Could you explain why it is needed? I mean, these functions should modify internal state of the lexer, but rather not the reader itself.

Reply:
The reader is passed as the lexer needs to be able to rewind it in certain rules. Using the bookmarks implies access to the reader itself.

Reply:
LEts talk about this design tomorrow! :)